src/v2/connectors/redisGraph.js

```diff
@@ -304,11 +305,35 @@ export default class RedisGraphConnector {
   async runApplicationsQuery() {
     const { withClause, whereClause } = await this.createWhereClause([], ['app']);
     const matchClause = `MATCH ${APPLICATION_MATCH} ${whereClause}`;
-    const returnClause = 'RETURN DISTINCT app._uid, app.name, app.namespace, app.created, app.dashboard, app.selfLink, app.label ORDER BY app.name, app.namespace ASC';
+    const returnClause = 'RETURN DISTINCT app._uid, app.name, app.namespace, app.created, app.apigroup, app.apiversion, app.dashboard, app.label, app.cluster ORDER BY app.name, app.namespace ASC';
     const query = `${withClause} ${matchClause} ${returnClause}`;
     return this.executeQuery({ query, removePrefix: false, queryName: 'runApplicationsQuery' });
   }
 
+  /*
+   * Get Argo Applications.
+   */
+  async runArgoApplicationsQuery() {
+    const { withClause, whereClause } = await this.createWhereClause([], ['app']);
+    const matchClause = `MATCH ${ARGO_APPLICATION_MATCH} ${whereClause}`;
+    const returnClause1 = 'RETURN DISTINCT app._uid, app.name, app.namespace, app.cluster, app.created, app.apigroup, app.apiversion, app.label,';
+    const returnClause2 = 'app.destinationName, app.destinationServer, app.destinationNamespace, app.repoURL, app.path, app.chart, app.targetRevision';
+    const orderBy = 'ORDER BY app.name, app.namespace, app.cluster ASC';
+    const query = `${withClause} ${matchClause} ${returnClause1} ${returnClause2} ${orderBy}`;
+    return this.executeQuery({ query, removePrefix: false, queryName: 'runArgoApplicationsQuery' });
+  }
+
+  /*
+   * Get Argo cluster secrets.
+   */
+  async runArgoClusterSecretsQuery() {
+    const { withClause, whereClause } = await this.createWhereClause([], ['s']);
+    const matchClause = `MATCH (s:Secret) ${whereClause} AND 'argocd.argoproj.io/secret-type=cluster' IN s.label`;
```

Comment:
You can get better performance if you change the order of this query to match the labels first.

----
In this scenario the `whereClause` is inefficient because we only want secrets, but `whereClause` contains all kinds.  **Fixing this will require a large change, so only mentioning it for your awareness of something we could improve in the future.**